server/modules/analysis/manager/index.js

```diff
@@ -66,3 +66,16 @@ export const persistAll = async ({ user, surveyId, chain, step = null, calculati
       throw new SystemError('appErrors.processingChainCannotBeSaved')
     }
   })
+
+export const importChain = async ({ user, surveyId, chain }, client = DB.client) =>
+  client.tx(async (tx) => {
+    await persistChain({ user, surveyId, chain }, tx)
+    const steps = Chain.getProcessingSteps(chain)
+
+    if (steps.length <= 0) return
+    await insertStepsInBatch({ user, surveyId, steps }, tx)
+    const calculations = []
+    steps.forEach((step) => (Step.getCalculations(step) || []).forEach((calculation) => calculations.push(calculation)))
```

Comment:
I think it could be more readable with something like:
`const calculations = steps.flatMap(Step.getCalculations)`